### PR TITLE
SQL statements that are of TextClause type need to be expressed as subqueries formally before being included in enclosing queries

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* If a `TextClause` query is to be part of a larger query, then it must be expressed as a `subquery()` formally; otherwise, the final SQL will contain syntax errors;
* Include handling for `Trino` (use subqueries instead of temporary tables, because the latter are not support in Trino);
* Minor cleanup.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- JIRA: GREAT-454
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 1]
```diff
@@ -480,6 +480,14 @@ def get_domain_records(
         else:
             selectable = data_object.selectable
 
+        """
+        If a custom query is passed, selectable will be TextClause and not formatted
+        as a subquery wrapped in "(subquery) alias". TextClause must first be converted
+        to TextualSelect using sa.columns() before it can be converted to type Subquery
+        """
```
Comment (NathanFarmer) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py:487]: Where did this comment come from? I wrote this comment at one point, and then @cdkini was working on a bugfix in this block I believe.

**Review by NathanFarmer — COMMENTED** [review]
One question - the rest looks good to me!

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py:487]: @NathanFarmer I copied it from another section in this module -- the description fit perfectly the current purpose (`RuntimeBatchRequest` containing an SQL query in plain text) and the code was needed in this new place, too.  Thanks.

**Review by Shinnnyshinshin — APPROVED** [review]
Looks great to me as well 🙇🏼

**Review by NathanFarmer — APPROVED** [review]
LGTM

**Pull request merged**